### DISCUSSION TITLE
How can i get this app working? I'm having problems in setup

### DISCUSSION THREAD

**Discussion by sumitt1080** [ketanchoyal/Our-E-School#32] [General]
Hi, i added a school  and login details, using which i can get to screen after login, where we need to enter profile details. But after i fill it all details, i can't get to next screen.

**Comment by kzawadi** [comment]
goos steps so far you just need to upload the firebase functions from the functions folder and put the URL provided after you upload those functions to server.dart

**Reply by sumitt1080** [reply]
Uploading function require paid plan, right? So when i upload those function then only it will work?

**Reply by kzawadi** [reply]
yes but ...upgrading to blaze plan won't cost if you stay within the limits...and you have 300 USD free usage I think ...and you won't be charged until you accept again to be charged (I mean the card you enter is just to activate)

**Reply by sumitt1080** [reply]
Okay, let me try. Thank you

**Reply by sumitt1080** [reply]
I tried deploying the function and ran following commands
npm install firebase-functions@latest firebase-admin@latest --save
npm install -g firebase-tools

but after this firebase.json is throwing error. Also, before all this, one thing is, when i view code of index.ts, it shows me 9+ errors. But when i close and move to other file, errors of index.ts disappered.

**Reply by sumitt1080** [reply]
Is the procedure to deploy typescript functions different from node functions?

**Reply by kzawadi** [reply]
make sure you connect the firebase project specifically the one you have firestore already configured so it can ready the firestore you use with this app ....  also install all the packages (in packages.json)    ... finally, the firebase documentation in uploading these function can guide properly   .... simple run is this 

1. create a firebase functions project and link it with the project which you use with our-e-school (firebase CLI will guide you through this )
2. copy the contents of the function folder (custom code only)
3. then upload

**Reply by sumitt1080** [reply]
can you tell me where i am getting wrong? 
![Capture](https://user-images.githubusercontent.com/51052011/122157449-7f353680-ce88-11eb-95a7-53d6969ed5ac.PNG)

![Capture1](https://user-images.githubusercontent.com/51052011/122157733-008cc900-ce89-11eb-85fa-2228b65e75da.PNG)

**Reply by kzawadi** [reply]
There are different ways .....  I suggest you create/initialize a firebase function project somewhere else in your disk then replace the code and package.json then fetch `npm install` ..... remember only copy src and packages.json
